src/main/java/frc/robot/subsystems/chooser/Chooser.java

```diff
@@ -0,0 +1,61 @@
+package frc.robot.subsystems.chooser;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import frc.robot.hardware.digitalinput.DigitalInputInputsAutoLogged;
+import frc.robot.hardware.digitalinput.IDigitalInput;
+import frc.robot.hardware.motor.IMotor;
+import frc.utils.GBSubsystem;
+import org.littletonrobotics.junction.Logger;
+
+public class Chooser extends GBSubsystem {
+
+	private final IMotor motor;
+	private final ChooserStuff chooserStuff;
+	private final ChooserCommandsBuilder commandBuilder;
+	private final IDigitalInput digitalInput;
+	private final DigitalInputInputsAutoLogged digitalInputInputs;
+
+	public Chooser(ChooserStuff chooserStuff) {
+		super(chooserStuff.logPath());
+		this.motor = chooserStuff.motor();
+		this.digitalInput = chooserStuff.digitalInput();
+		this.chooserStuff = chooserStuff;
+		this.digitalInputInputs = new DigitalInputInputsAutoLogged();
+
+		this.commandBuilder = new ChooserCommandsBuilder(this);
+		updateInputs();
+	}
+
+	public ChooserCommandsBuilder getCommandBuilder() {
+		return commandBuilder;
+	}
+
+	public boolean isObjectIn() {
+		return digitalInputInputs.debouncedValue;
+	}
+
+	public void updateInputs() {
```

Comment:
are signals inputs?

Reply:
i think so

Reply:
signals are a custom object we have made, inputs are automatic stuff that is part of another library. Signals aren't inputs, but also Romy wrote this func with this name

Reply:
Just do it like in funnel `update()`

Reply:
Come on opposing opinions.. Yoav already said those are like inputs and I should name the function updateinputs

Reply:
I also need to change funnel to update inputs..